spec/models/asset_spec.rb

```diff
@@ -256,69 +256,27 @@
 
     before do
       allow_any_instance_of(VirusScanner).to receive(:clean?).and_return(true)
+      allow(SaveToCloudStorageWorker).to receive(:perform_async)
     end
 
     it 'schedules saving the asset to cloud storage' do
-      expect {
-        asset.scan_for_viruses
-      }.to change(Delayed::Job, :count).by(1)
+      expect(SaveToCloudStorageWorker).to receive(:perform_async).with(asset)
 
-      expect(most_recently_enqueued_job).to have_payload(asset)
-      expect(most_recently_enqueued_job).to have_method_name(:save_to_cloud_storage)
+      asset.scan_for_viruses
     end
   end
 
   describe "#save_to_cloud_storage" do
-    let(:asset) { FactoryGirl.create(:clean_asset) }
-
-    context 'when S3 bucket is configured' do
-      let(:cloud_storage) { double(:cloud_storage) }
-
-      before do
-        allow(Services).to receive(:cloud_storage).and_return(cloud_storage)
-      end
-
-      it 'saves the asset to cloud storage' do
-        expect(cloud_storage).to receive(:save).with(asset)
-
-        asset.save_to_cloud_storage
-      end
-
-      context 'when an exception is raised' do
-        let(:exception_class) { Class.new(StandardError) }
-        let(:exception) { exception_class.new }
-
-        before do
-          allow(cloud_storage).to receive(:save).and_raise(exception)
-        end
-
-        it 'reports the exception to Errbit via Airbrake' do
-          expect(Airbrake).to receive(:notify_or_ignore)
-            .with(exception, params: { id: asset.id, filename: asset.filename })
-
-          asset.save_to_cloud_storage rescue exception_class
-        end
-
-        it 're-raises the exception so Delayed::Job will re-try it' do
-          allow(Airbrake).to receive(:notify_or_ignore)
+    let(:asset) { FactoryGirl.create(:asset) }
```

Comment:
Minor: Does this asset need to be saved in the database?

Reply:
I suspect that maybe it didn't need to be saved in the database in this commit, but does need to be in the next commit when the asset ID is used. So I suspect you can ignore my previous comment.

Reply:
Good catch - I don't think I'll edit the previous commit to fix this, but I think you're right that I don't need a persisted asset in that commit.